File: lib/core/models/playlist.dart
class Playlist {
  final int id;
  final String? userId;
  final String name;
  final String? description;
  final String? coverUrl;
  final String playlistType;
  final bool isSystem;
  final bool isPublic;
  final int songsCountCache;

  const Playlist({
    required this.id,
    this.userId,
    required this.name,
    this.description,
    this.coverUrl,
    required this.playlistType,
    required this.isSystem,
    required this.isPublic,
    this.songsCountCache = 0,
  });

  factory Playlist.fromMap(Map<String, dynamic> map) {
    return Playlist(
      id: map['id'] is int ? map['id'] : int.tryParse(map['id'].toString()) ?? 0,
      userId: map['user_id'] as String?,
      name: map['name'] ?? '',
      description: map['description'] as String?,
      coverUrl: map['cover_url'] as String?,
      playlistType: map['playlist_type'] ?? 'user',
      isSystem: map['is_system'] ?? false,
      isPublic: map['is_public'] ?? false,
      songsCountCache: map['songs_count_cache'] ?? 0,
    );
  }
}

File: Backend/Models/Playlist.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MusicBackend.Models
{
    [Table("playlists")]
    public class Playlist
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("name")]
        [System.Text.Json.Serialization.JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [Column("description")]
        [System.Text.Json.Serialization.JsonPropertyName("description")]
        public string? Description { get; set; }

        [Column("cover_url")]
        [System.Text.Json.Serialization.JsonPropertyName("cover_url")]
        public string? CoverUrl { get; set; }

        [Column("playlist_type")]
        [System.Text.Json.Serialization.JsonPropertyName("playlist_type")]
        public string PlaylistType { get; set; } = "user";

        [Column("is_system")]
        [System.Text.Json.Serialization.JsonPropertyName("is_system")]
        public bool IsSystem { get; set; } = false;

        [Column("user_id")]
        [System.Text.Json.Serialization.JsonPropertyName("user_id")]
        public Guid? UserId { get; set; }

        [Column("created_at")]
        [System.Text.Json.Serialization.JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("is_public")]
        [System.Text.Json.Serialization.JsonPropertyName("is_public")]
        public bool IsPublic { get; set; } = false;

        [Column("songs_count_cache")]
        [System.Text.Json.Serialization.JsonPropertyName("songs_count")]
        public int SongsCount { get; set; } = 0;

        [Column("saves_count_cache")]
        [System.Text.Json.Serialization.JsonPropertyName("saves_count")]
        public int SavesCount { get; set; } = 0;

        [Column("total_listens_cache")]
        [System.Text.Json.Serialization.JsonPropertyName("total_listens")]
        public int TotalListens { get; set; } = 0;

        [NotMapped]
        [System.Text.Json.Serialization.JsonPropertyName("song_ids")]
        public List<int> SongIds { get; set; } = new List<int>();
    }
}
